dynamiqs/progress_bar.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from typing import Callable, Union
```

Comment:
New-gen typing, we probably don't need this

Callable -> callable
Union[a, b] -> a | b

Reply:
This is why we `from __future__ import annotations`

Reply:
TIL